lib/src/adaptiveWidgets/adaptive_icon.dart

```diff
@@ -2,30 +2,177 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 
 import 'adaptive_widget.dart';
+import 'dart:io';
+
+enum IconDataSet {
```

Comment:
I'm not sure if IconDataSet is the best name here. What are you thinking about e.g. IconSource or IconValue or something similar?